vault/vault_secret_client.go

```diff
@@ -0,0 +1,150 @@
+package vault
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/rs/zerolog"
+)
+
+type VaultClient struct {
+	address     string
+	authPath    string
+	role        string
+	jwtFilePath string
+	logger      *zerolog.Logger
+}
+
+const ROLE = "role"
+const JWT = "jwt"
+
+func getFullAuthPath(authPath string) string {
+	if authPath == "" {
+		return ""
+	}
+	fullAuthPath := fmt.Sprintf("/v1/auth/%s/login", authPath)
+	return fullAuthPath
+}
+
+// return structure for Vault Client, based on configuration
+func NewVaultClient(conf map[interface{}]interface{}, logger *zerolog.Logger) VaultClient {
+	address := "http://vault.fybrik-system:8200"
```

Comment:
should it be env variable ?

Reply:
removed default value for 'address' and other vault configuration fields